src/app/g3w-ol/controls/screenshotcontrol.js

```diff
@@ -27,12 +27,15 @@ proto.change = function(layers=[]){
  * same origin URL of current application in order to avoid
  * CORS issue while getting map image.
  * 
+ * Layers that don't have a source URL are excluded (eg. base layers)
+ * 
  * @param {array} layers
  * 
  * @returns {boolean}
  */
 proto.checkVisible = function(layers=[]) {
-  return undefined === layers.find((layer) => !sameOrigin(layer.getSource() && layer.getSource().url, location));
+  return undefined === layers.find((layer) => !sameOrigin(layer.getSource && layer.getSource() && layer.getSource().url, location));
```

Comment:
@volterra79 What are you trying to fix with this edit https://github.com/g3w-suite/g3w-client/pull/426/commits/3ca0c248ffc1e8850de2a8a10284c44e8493fa78?

Written as above isn't it safe enough?